samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs

```diff
@@ -123,5 +123,53 @@ public async Task AssignmentsEvolvableEnumsAsync()
                 Console.WriteLine($"AssignmentsEvolvableEnumsAsync: {ex.ToString()}");
             }
         }
-    }
+
+        /// <summary>
+        /// Workflow to create Assignment resources under the assignment resource folder
+        /// </summary>
+        /// <param name="appOnly">True value authenticates the graph client with application permissions only, otherwise it will be created with delegated permissions.</param> 
+        public async Task AssignmentResourceAsync(bool appOnly = false)
+        {
+            try
+            {
+                string assignmentId = string.Empty;
+
+                // Get a Graph client based on the appOnly parameter
+                var graphClient = appOnly ? GraphClient.GetApplicationClient(_config["tenantId"], _config["appId"], _config["secret"]) : GraphClient.GetDelegateClient(_config["tenantId"], _config["appId"], _config["teacherAccount"], _config["password"]);
+
+                // Create assignment
+                var assignment = await Assignment.CreateSampleAssignmentAsync(graphClient, _config["classId"]);
+                assignmentId = assignment.Id;
+                Console.WriteLine($"Assignment created successfully with Id: {assignment.Id} in state: {assignment.Status}");
+
+                // Set Up Assignment Resources Folder
+                await Assignment.SetupResourcesFolder(graphClient, _config["classId"], assignmentId);
```

Comment:
SetupResourcesFolderAsync

Reply:
Changed "SetupResourcesFolder" to "SetupResourcesFolderAsync"